### PR TITLE
[SC-32] [#4] Feature Mandatory fields

### PR DESCRIPTION
**Pull request by sutrixtuanvothanh**
Ticket: SC-32, #4 Create Layout And Validation Logic

 

**Detailed purpose of the PR**

- Feature Mandatory fields, 
- To create or modify a cohort, it should at least contain one provider and one content option.
Add a mandatory logic, so that the "submit" button is not active until one option in each column haven't been selected
[Mandatory fields]

**Result and observation**
Create new cohort
featuremandatory
Create new cohort ready 
mandatory-create-valid
Edit invalid
mandatory-invalid
Edit Valid
mandatory-valid

- [ ]  **Breaking changes ?**  
If checked, what have you broken ?

- [x] **Extra lib ?** 
If checked, Which extra lib did you add ? (name, purpose, link ...).

**How tested ?**
 - [x] Already covered by tests
 - [x] Manual testing
 - [x] Unit testing

### Checklist self-review:

* [x] My code follows the style guidelines of this project
* [ ] Merged with latest code from develop and resolved conflict
* [x] I have checked other pages which use the same component as well
* [x] I have tested my updated code in Firefox
* [x] I have tested my updated code in Edge
* [x] I have tested my updated code in Safari
* [x] My function and variable names are meaningful
* [x] My function and variable names are correct spelling
* [ ] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation (Jira, Confluence)
* [x] My changes generate no eslint warnings or errors
* [x] I have added unit test that prove my fix is effective or that my feature works
* [x] I have added automation test that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by sutrixtuanvothanh** [comment]
[Mandatory fields](https://trello.com/c/lGz22ISc/4-mandatory-fields)

File: packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js [hunk 1]
```diff
@@ -32,15 +32,18 @@ const InputSplitScreen = props => {
     <div>
       <div className={style.splitDefault}>
         <div className={style.splitLeft}>
-          <p className={style.title}>{leftSection.title}</p>
+          <p className={style.title}>{leftSection.title}*</p>
```
Comment (esa-coorp) [inline packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js]: include a `mandatory` (or  `required`) field on props for manage `*`

File: packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js [hunk 2]
```diff
@@ -32,15 +32,18 @@ const InputSplitScreen = props => {
...
           {criteriaList}
         </div>
         <div className={style.splitRight}>
-          <p className={style.title}>{rightSection.title}</p>
+          <p className={style.title}>{rightSection.title}*</p>
```
Comment (esa-coorp) [inline packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js]: same

File: packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js [hunk 3]
```diff
@@ -32,15 +32,18 @@ const InputSplitScreen = props => {
...
           {collectionList}
         </div>
       </div>
-      <div className={style.buttonGroup}>{buttonList}</div>
+      <div className={style.rowBottom}>
+        <div className={style.buttonGroup}>{buttonList}</div>
+        <p className={style.mandatoryFields}>*Mandatory fields</p>
```
Comment (esa-coorp) [inline packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js]: use translate function

File: packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js (hunk 1 above)
Comment (sutrixtuanvothanh) [reply packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js]: done

File: packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js (hunk 2 above)
Comment (sutrixtuanvothanh) [reply packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js]: done

File: packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js (hunk 3 above)
Comment (sutrixtuanvothanh) [reply packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js]: done

File: packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js [hunk 4]
```diff
@@ -32,22 +34,39 @@ const InputSplitScreen = props => {
...
 InputSplitScreen.propTypes = {
-  leftSection: PropTypes.shape({}),
-  rightSection: PropTypes.shape({})
+  leftSection: PropTypes.shape({
+    title: PropTypes.string,
+    values: PropTypes.array,
+    required: PropTypes.bool
+  }),
+  rightSection: PropTypes.shape({
+    title: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (sutrixtuanvothanh) [reply packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js]: done

File: packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js [hunk 5]
```diff
@@ -32,22 +34,39 @@ const InputSplitScreen = props => {
...
+    values: PropTypes.array,
```
[the remark this answers is not in the document]
Comment (sutrixtuanvothanh) [reply packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js]: done

File: packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js [hunk 6]
```diff
@@ -32,22 +34,39 @@ const InputSplitScreen = props => {
...
+        <p className={style.mandatoryFields}>{translate('mandatory_fields')}</p>
+      </div>
     </div>
   );
 };
 
 InputSplitScreen.propTypes = {
-  leftSection: PropTypes.shape({}),
-  rightSection: PropTypes.shape({})
+  leftSection: PropTypes.shape({
```
[the remark this answers is not in the document]
Comment (sutrixtuanvothanh) [reply packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js]: done

**Pull request merged**